packages/runtime-common/indexer.ts

```diff
@@ -499,6 +504,144 @@ export class Indexer {
     ];
   }
 
+  private buildPrerenderedInstancesQuery(everyCondition: CardExpression) {
+    // For evety indexed card instance (type = instance), go through its deps and find the corresponding CSS index entries (type = css)
+    return [
+      `WITH instance_records AS (
+        SELECT i.file_alias, i.deps, i.embedded_html ->> 'default' as default_embedded_html, i.search_doc, i.url, r.realm_url
+        FROM boxel_index AS i ${tableValuedFunctionsPlaceholder}
+        INNER JOIN realm_versions r ON i.realm_url = r.realm_url`,
+      'WHERE',
+      ...everyCondition,
+      `),
+      deps_expanded AS (
+        SELECT
+          ir.file_alias AS instance_file_alias,
+          ir.default_embedded_html AS instance_embedded_html,
+          ir.realm_url,
+          ir.search_doc,
+          ir.url,
+          jsonb_array_elements_text(ir.deps) AS dep
+        FROM instance_records ir
+      ),
+      css_matches AS (
+        SELECT DISTINCT
+          d.instance_file_alias,
+          d.instance_embedded_html,
+          d.search_doc,
+          d.realm_url,
+          d.url,
+          b.file_alias AS css_file_alias,
+          b.source AS css_source
+        FROM deps_expanded d
+        LEFT JOIN boxel_index b ON b.file_alias = d.dep AND b.type = 'css'
+      ),
+      aggregated_results AS (
+        SELECT
+          instance_file_alias,
+          MAX(instance_embedded_html) AS instance_embedded_html,
+          MAX(realm_url) AS realm_url,
+          MAX(url) as url,
+		      (ARRAY_AGG(search_doc))[1] AS search_doc,
+          COALESCE(array_agg(css_file_alias ORDER BY css_file_alias) FILTER (WHERE css_file_alias IS NOT NULL), ARRAY[]::text[]) AS css_file_alias_array,
+          COALESCE(array_agg(css_source ORDER BY css_file_alias) FILTER (WHERE css_file_alias IS NOT NULL), ARRAY[]::text[]) AS css_source_array
+        FROM css_matches
+        GROUP BY instance_file_alias
+      )`,
+    ];
+  }
+
+  async searchPrerendered(
+    realmURL: URL,
+    { filter, sort, page }: Query,
+    loader: Loader,
+    opts?: QueryOptions,
+  ): Promise<{
+    prerenderedCards: PrerenderedCard[];
+    meta: QueryResultsMeta;
+  }> {
+    let version: number;
+    if (page?.realmVersion) {
+      version = page.realmVersion;
+    } else {
+      let currentRealmVersion = await this.fetchCurrentRealmVersion(realmURL);
+      version = opts?.useWorkInProgressIndex
+        ? currentRealmVersion + 1
+        : currentRealmVersion;
```

Comment:
you shouldn't have to do this. the conditions for the realmVersionExpression do this for you. You have 3 options for this:
1 a specific realm version
2. production realm version
3. the WIP realm version. 

use pass the opts.useWorkInProgressIndex to the `realmVersionExpression`. no need to fetch the current realm version. that is handled internally in the SQL

Reply:
oh wait--I take that back i see what is going on here--this is used to hold the realm version stable while the user pages thru the results. never mind--this is the same approach that we do for the main search.